src/atom_finder/classifier/infix-operator-precedence.clj

```diff
@@ -0,0 +1,37 @@
+(in-ns 'atom-finder.classifier)
+(import '(org.eclipse.cdt.core.dom.ast IASTBinaryExpression IASTConditionalExpression IASTUnaryExpression)
+        '(org.eclipse.cdt.internal.core.dom.parser.cpp CPPASTConditionalExpression))
+
+(defn non-associative?
+  "Returns true if a non-associative operator, returns nil otherwise"
+  [node]
+  (let [non-associatives
+        #{ ;Binary operations that are non-associative
+          IASTBinaryExpression/op_minus IASTBinaryExpression/op_divide IASTBinaryExpression/op_modulo
+          IASTBinaryExpression/op_shiftLeft IASTBinaryExpression/op_shiftRight IASTBinaryExpression/op_greaterThan 
+          IASTBinaryExpression/op_greaterEqual IASTBinaryExpression/op_lessThan IASTBinaryExpression/op_lessEqual CPPASTConditionalExpression}]
+    (contains? non-associatives (if (instance? IASTBinaryExpression node) (.getOperator node) (type node)))))
+
+(defn operator-equal?
+  "compare types two nodes, if both binary expression then compare their operators"
+  [& nodes]
+  (if (or (every? #(instance? IASTBinaryExpression %) nodes) (every? #(instance? IASTUnaryExpression %) nodes))
+    (apply = (map (memfn getOperator) nodes))
+    (apply = (map type nodes))))
+
+
+(defn infix-operator-precedence-atom?
+  "Is this node a infix-operator-precedence-atom?"
+  [node]
+  (and
+   (and (precedence-level node)
+        (not (assignment? node)))
+
+   (if (non-associative? node)
+     ;;Current node is a non-associative operator
+     (some #(precedence-level %) (children node))
```

Comment:
(precedence-level node) is basically testing whether or not this node represents an operator. If it returns a number then yes, if nil then this node can be ignored. It was written initially to make sure the node has .getOperator function before actually calling it. But I feel like operator-equal? function basically takes care of that case. I'll do more testing.

With unary operators taken into account I think the code needs to be reconsidered.

`! a && b`: we agreed that this code is confusing, and currently my code takes care of it
`*p && b`: assuming p is pointing to a boolean, I don't think this code is as confusing, but in terms of precedence level it is basically the same as the one above
`++a + b`: again, basically the same thing, but I don't think it's as confusing as the first one

I'm not sure what you're going to add and it really depends on how eclipse parses and what we consider confusing or not confusing. But I think it'll maintain a good accuracy.